docs/02-for-app-authors/03-appdata-guidelines/01-quality-guidelines.md

```diff
@@ -0,0 +1,75 @@
+# Quality guidelines
+
+For the quick guidelines for the actual content (descriptions, screenshots), see the Appstream [docs](https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html).
+
+The following guidelines are optional, but recommended. We will likely promote apps that follow them in certain places.
+
+## General
+
+### No trademark violations
+
+Your app should not violate any trademarks. This includes the name, icon, and summary.
+
+For example, if your app is a Telegram client or web wrapper, it should not be called `Telegram for Linux` or use the official Telegram icon. Instead, it should have its own identity (e.g. `Paper Plane`, with a custom icon). If it's made clear that it's a third party client it can be ok to mention the original service (e.g. `Tally for Plausible`), but in most cases it should be avoided.
+
+:::tip
+If you're not sure if your app violates any trademarks, feel free to ask us on [Matrix](https://matrix.to/#/#flathub:matrix.org).
+:::
+
+### Managed by app author
+
+We recommend to verify your app on Flathub, but it’s not required.
+
+## App Icon
+
+### Icon size
+
+The icon should be SVG, or a PNG with a size greater or equal to 256x256px. It has to be square.
+
+### Icon ratio
+
+The icon shouldn't fill less than 90% of the canvas or touch the edges.
+
+### Good contrast
+
+The icon should have good contrast on dark and light backgrounds. In particular, avoid single-color glyphs and the like since they can be very hard to see if the background color is similar. Dark colors near the edges are also a problem, since shadows don't help much with contrast on dark backgrounds.
+
+### Not too much detail
+
+The icon should be recognizable at small sizes.
+
+### No baked-in shadows
+
+The icon should not have any baked-in shadows or glows. Both the Flathub website and native stores apply a shadow to all icons, so baked-in shadows result in double shadows or other glitches.
+
+### In line with style
+
+The icon should be in line with the contemporary icon style.
+
+## App Name
+
+### Not too long
+
+The name should ideally be no longer than 15 characters, and must be shorter than 20 characters.
+
+### Just the name
+
+The name should be just the name, without any additional info. For example, don't append what the app does to the name (e.g. "Apostrophe - Markdown Editor"), and use the summary to provide this extra information instead.
+
+### No weird formatting
+
+The name should not have any weird formatting or punctuation.
```

Comment:
```suggestion
The name should not have any weird formatting or punctuation. For example, it should not be all-lowercase, all-uppercase, camel case, or contain dashes or periods.
```

Reply:
I can't agree on `all-lowercase`, as somebody working at a company that has a stylized lowercase name and we usually also extend that style to our (internal) apps.

Reply:
It's one thing when it's part of a brand, but usually it's just a mistake:

![image](https://github.com/flathub/documentation/assets/1908896/9ae1d4c3-3db5-4c43-87d3-15272e878589)

Practically speaking do you have examples of apps where you think lowercase is appropriate?

Reply:
https://flathub.org/apps/collection/popular/1 actually has multiple examples

I've included some camel case and uppercase too

![image](https://github.com/flathub/documentation/assets/5943908/f0e55660-506e-4905-b427-b32577506f8e)

![image](https://github.com/flathub/documentation/assets/5943908/cc26d694-a33c-4666-a8ad-bbc7132bb3dc)

![image](https://github.com/flathub/documentation/assets/5943908/f2a19f32-a23c-4d24-8d7f-6348114aeb94)

![image](https://github.com/flathub/documentation/assets/5943908/7afc18e5-f5f3-4a32-8e71-d4e7e9ae76aa)

![image](https://github.com/flathub/documentation/assets/5943908/f503a683-9a79-4758-91c1-6ea4b01c7038)

![image](https://github.com/flathub/documentation/assets/5943908/bdf857af-d9a9-4098-821d-56f2651df13b)

![image](https://github.com/flathub/documentation/assets/5943908/691ff2dc-cc9a-4749-a98c-3f45fd27f3cf)

Reply:
Hmm yeah, fair enough. I guess we could make an exception for cases where it's part of the brand?
```suggestion
The name should not have any weird formatting or punctuation. For example, it should not be all-lowercase, all-uppercase, camel case, or contain dashes or periods. Cases where the weird formatting is part of the brand are exempt from this (e.g. "VLC").
```

Reply:
Yeah, I would be fine with that, but it might be abused - which we can see about then.